impress-org/givewp
# Fix: Prevent notice about 'decimals_enabled' when adding/updating forms

Resolves #6261 

## Description


Notices were being generated when adding/updating forms that said `Undefined index: decimals_enabled`. This was happening because of some missing checks in the backwards compatibility for the form options. This PR fixes that by using `isset()` to check for the deeply nested properties of the `$settings` array before we assign them.

## Affects


Shouldn't affect anything else, just adding some guardrails around existing functionality.

## Testing Instructions


1. Install GiveWP on a fresh WP site
2. With debugging enabled, create a new form and publish it
3. Make sure no notices about `decimals_enabled` appear on the frontend or in the `debug.log` file 

- OR -

1. Select a form. Go into the `give_form_meta` table, find the row with `meta_key` of `_give_{template name}_form_template_settings`, and delete the `decimals_enabled` property.
2. In the WP admin, go to Donations -> All Forms and edit the form from step 1
3. Update the form
4. Make sure no notices about `decimals_enabled` appear on the frontend or in the `debug.log` file

## Pre-review Checklist
